src/score.jl

```diff
@@ -0,0 +1,473 @@
+    result[:star_color_gr] = objs[:psfmag_g] .- objs[:psfmag_r]
+    result[:star_color_ri] = objs[:psfmag_r] .- objs[:psfmag_i]
+    result[:star_color_iz] = objs[:psfmag_i] .- objs[:psfmag_z]
+
+    # gal colors
+    result[:gal_color_ug] = gal_mag_u .- gal_mag_g
+    result[:gal_color_gr] = gal_mag_g .- gal_mag_r
+    result[:gal_color_ri] = gal_mag_r .- gal_mag_i
+    result[:gal_color_iz] = gal_mag_i .- gal_mag_z
+
+    # gal shape
+    result[:gal_fracdev] = objs[:fracdev_r]
+    result[:gal_ab] = where(usedev, objs[:devab_r], objs[:expab_r])
+    result[:gal_angle] = where(usedev, objs[:devphi_r], objs[:expphi_r])
+    result[:gal_scale] = where(usedev, objs[:devrad_r], objs[:exprad_r])
+
+    return result
+end
+
+
+"""
+Convert two fluxes to a color: mag(f1) - mag(f2), assuming the same zeropoint.
+Returns NaN if either flux is nonpositive.
+"""
+function fluxes_to_color(f1::Real, f2::Real)
+    (f1 <= 0. || f2 <= 0.) && return NaN
+    return -2.5 * log10(f1 / f2)
+end
+@vectorize_2arg Real fluxes_to_color
+
+
+"""
+load_primary(dir, run, camcol, field)
+
+Load the SDSS photObj catalog used to initialize celeste, and reformat column
+names to match what the rest of the scoring code expects.
+"""
+function load_primary(dir, run, camcol, field)
+
+    objs = SDSS.load_catalog_df(dir, run, camcol, field)
+
+    usedev = objs[:frac_dev] .> 0.5  # true=> use dev, false=> use exp
+    gal_flux_u = where(usedev, objs[:devflux_u], objs[:expflux_u])
+    gal_flux_g = where(usedev, objs[:devflux_g], objs[:expflux_g])
+    gal_flux_r = where(usedev, objs[:devflux_r], objs[:expflux_r])
+    gal_flux_i = where(usedev, objs[:devflux_i], objs[:expflux_i])
+    gal_flux_z = where(usedev, objs[:devflux_z], objs[:expflux_z])
+
+
+    result = DataFrame()
+    result[:objid] = objs[:objid]
+    result[:ra] = objs[:ra]
+    result[:dec] = objs[:dec]
+    result[:is_star] = objs[:is_star]
+    result[:star_flux_r] = objs[:psfflux_r]
+    result[:gal_flux_r] = gal_flux_r
+
+    # star colors
+    result[:star_color_ug] = fluxes_to_color(objs[:psfflux_u], objs[:psfflux_g])
+    result[:star_color_gr] = fluxes_to_color(objs[:psfflux_g], objs[:psfflux_r])
+    result[:star_color_ri] = fluxes_to_color(objs[:psfflux_r], objs[:psfflux_i])
+    result[:star_color_iz] = fluxes_to_color(objs[:psfflux_i], objs[:psfflux_z])
+
+    # gal colors
+    result[:gal_color_ug] = fluxes_to_color(gal_flux_u, gal_flux_g)
+    result[:gal_color_gr] = fluxes_to_color(gal_flux_g, gal_flux_r)
+    result[:gal_color_ri] = fluxes_to_color(gal_flux_r, gal_flux_i)
+    result[:gal_color_iz] = fluxes_to_color(gal_flux_i, gal_flux_z)
+
+    # gal shape
+    result[:gal_fracdev] = objs[:frac_dev]
+    result[:gal_ab] = where(usedev, objs[:ab_dev], objs[:ab_exp])
+
+    # TODO: the catalog contains both theta_[exp,dev] and phi_[exp,dev]!
+    # which do we use?
+    #result[:gal_angle] = where(usedev, objs[:theta_dev], objs[:theta_exp])
+    result[:gal_angle] = fill(NaN, size(objs, 1))
```

Comment:
Is `theta_[exp,dev]` scale and `phi_[exp,dev]` angle? That would resolve both ambiguities at once. Otherwise I think we'd have to ask Dustin.